test/gouda/fiber_worker_test.rb

```diff
@@ -0,0 +1,322 @@
+# frozen_string_literal: true
+
+require "gouda/test_helper"
+
+class FiberWorkerTest < ActiveSupport::TestCase
+  include AssertHelper
+
+  # Path for test file output similar to worker_test.rb pattern
+  PATH_TO_FIBER_TEST_FILE = -> { File.expand_path(File.join("tmp", "#{Process.pid}-gouda-fiber-worker-test-output.bin")) }
+
+  class FiberTestJob < ActiveJob::Base
+    self.queue_adapter = :gouda
+
+    def perform(value = "F")
+      File.open(PATH_TO_FIBER_TEST_FILE.call, "a") do |f|
+        f.write(value)
+      end
+    end
+  end
+
+  class FiberJobWithQueue < ActiveJob::Base
+    self.queue_adapter = :gouda
+    queue_as :fiber_queue
+
+    def perform
+      File.open(PATH_TO_FIBER_TEST_FILE.call, "a") do |f|
+        f.write("Q")
+      end
+    end
+  end
+
+  setup do
+    # Clean up test file before each test
+    File.delete(PATH_TO_FIBER_TEST_FILE.call) if File.exist?(PATH_TO_FIBER_TEST_FILE.call)
+  end
+
+  teardown do
+    # Clean up test file after each test
+    File.delete(PATH_TO_FIBER_TEST_FILE.call) if File.exist?(PATH_TO_FIBER_TEST_FILE.call)
+  end
+
+  test "fiber worker can be configured" do
+    original_use_fiber_scheduler = Gouda.config.use_fiber_scheduler
+    original_fiber_worker_count = Gouda.config.fiber_worker_count
+    original_async_db_pool_size = Gouda.config.async_db_pool_size
+
+    Gouda.configure do |config|
+      config.use_fiber_scheduler = true
+      config.fiber_worker_count = 5
+      config.async_db_pool_size = 10
```

Comment:
Why do we need a separate pool size when using fibers. and why should Gouda be controlling it?